scipy/optimize/_nonlin.py

```diff
@@ -1492,9 +1492,32 @@ def __init__(self, rdiff=None, method='lgmres', inner_maxiter=20,
             self.method_kw.setdefault('store_outer_Av', False)
             self.method_kw.setdefault('atol', 0)
 
+        valid_inner_params = dict(
+            bicgstab=["rtol", "atol"],
+            gmres=["rtol", "atol", "restart"],
+            lgmres=[
+                "rtol",
+                "atol",
+                "inner_m",
+                "outer_k",
+                "outer_v",
+                "store_outer_Av",
+                "prepend_outer_v"
+            ],
+            cgs=["rtol", "atol", "show", "check"],
+            minres=["rtol", "shift"],
+            tfqmr=["rtol", "atol"],
+        ).get(method, method)
+
         for key, value in kw.items():
             if not key.startswith('inner_'):
                 raise ValueError(f"Unknown parameter {key}")
+            if key[6:] not in valid_inner_params:
```

Comment:
You also need to accommodate for the possibility that `method` is a user-provided callable (no need to give any warnings here)